### PR TITLE
Add JSON5 parsing capabilities for integration configs

### PR DESCRIPTION
**Pull request by devesh-2002**
### Description
Integrated JSON5 

### Issues Resolved
Issue #1680 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by RyanL1997 — COMMENTED** [review]
Hi @devesh-2002, thank you for taking this on! In generally this is looking good to me, however, could you also sign off for your commit? 
```
git commit -s -m ""
```

**Comment by devesh-2002** [comment]
Is this okay?

**Comment by Swiddis** [comment]
From the DCO action [details](https://github.com/opensearch-project/dashboards-observability/pull/1732/checks?check_run_id=24065041721), signing off has 3 steps:

> 1. Ensure you have a local copy of your branch by [checking out the pull request locally via command line](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally).
> 2. In your local branch, run: `git rebase HEAD~4 --signoff`
> 3. Force push your changes to overwrite the branch: `git push --force-with-lease origin devesh1`

As far as the PR goes: This looks good at a glance, thanks for contributing! You might also consider adding some unit tests (or modifying existing ones) to cover the new functionality. I'll be available later this week to get this merged/backported.

**Comment by devesh-2002** [comment]
Hello. I have signed off the commits and I have also tried to add some tests. Could you please check it? @Swiddis

File: server/adaptors/integrations/__test__/local_fs_repository.test.ts
Comment (Swiddis) [inline server/adaptors/integrations/__test__/local_fs_repository.test.ts:116]: I'm not sure I follow what these unit tests are for, it's checking every file in a directory scan if they're readable? The second one might always fail since `expect(result.ok).toBe(false)` will fail any test where any file is readable.

On second thought I think it would be enough to test by updating one or two of the integrations in `repository` to just use json5 conventions, we already have tests that are equivalent to "all integrations in `repository` are readable" and if json5 isn't working it should be caught by those tests or integration tests when updating configs in the repository.

We also might need to update the config file search to handle both json5 and json file extensions, [currently it only checks for json](https://github.com/opensearch-project/dashboards-observability/blob/44727c33e55847da6d08ad69cd7bd48c1fd15fc4/server/adaptors/integrations/repository/integration_reader.ts#L155-L158). If you're willing to work on adding the ability to check both, feel free, but it's not critical to have it now (can just have json file extension with json5 data) so don't feel pressured.

**Review by Swiddis — CHANGES_REQUESTED** [review]
Thanks for adding DCO and tests. I enabled the CI workflows for the PR to help narrow the feedback loop, but I think there's still some more changes to look at. Thanks again for contributing!

File: server/adaptors/integrations/__data__/repository/aws_vpc_flow/assets/vpc_live_all_mv-1.0.0.sql
Comment (devesh-2002) [reply server/adaptors/integrations/__data__/repository/aws_vpc_flow/assets/vpc_live_all_mv-1.0.0.sql:1]: I am not sure why this has happened. I did not change anything in that file.

File: server/adaptors/integrations/__test__/local_fs_repository.test.ts
Comment (devesh-2002) [reply server/adaptors/integrations/__test__/local_fs_repository.test.ts:116]: > On second thought I think it would be enough to test by updating one or two of the integrations in repository to just use json5 conventions

So do you mean that I should remove the unit tests and update the `fs_data_adaptor` in `repository`? I am a bit confused regarding how do I update

File: server/adaptors/integrations/__data__/repository/aws_vpc_flow/assets/vpc_live_all_mv-1.0.0.sql
Comment (RyanL1997) [reply server/adaptors/integrations/__data__/repository/aws_vpc_flow/assets/vpc_live_all_mv-1.0.0.sql:1]: I think this happened after the rebase of one of the out synced changes: https://github.com/opensearch-project/dashboards-observability/pull/1725. Let's just revert this back.

File: server/adaptors/integrations/__test__/local_fs_repository.test.ts
Comment (Swiddis) [reply server/adaptors/integrations/__test__/local_fs_repository.test.ts:116]: I mean updating integration configs in `__data__/repository`, e.g. `nginx-1.0.0.json` or `apache-1.0.0.json`, to use json5. Tests will catch if files in `__data__/repository` are malformed so updating something there will work to test if json5 works.

**Comment by devesh-2002** [comment]
Hello,@Swiddis. Can you check if it is correct now?

**Comment by Swiddis** [comment]
You can run tests locally with `yarn test`, you don't need to wait for the CI here to get approved (we can't approve the CI here if there's merge conflicts). Running `yarn test integrations`:

```
$ yarn test integrations
...
Test Suites: 4 failed, 19 passed, 23 total
Tests:       18 failed, 136 passed, 154 total
Snapshots:   15 passed, 15 total
Time:        81.812 s
```

In particular the tests in `integrations/__test__/local_fs_repository.test.ts` will be more informative since they all have to do with the local files you're modifying (and the other tests depend on them passing):

```
$ yarn test integrations/__test__/local_fs_repository
Test Suites: 1 failed, 1 total
Tests:       3 failed, 1 passed, 4 total
Snapshots:   0 total
Time:        4.588 s, estimated 72 s
```

In this case it's because the file extension was changed to json5, but there's [no handling for json5 filenames](https://github.com/opensearch-project/dashboards-observability/blob/44727c33e55847da6d08ad69cd7bd48c1fd15fc4/server/adaptors/integrations/repository/integration_reader.ts#L155-L158). You can try to implement this handling in the code if you'd like, or you can change the extensions back to json but keep the json5-formatted data here.

**Comment by devesh-2002** [comment]
I have changed the extensions to back to JSON keeping the data as JSON5.

**Comment by Swiddis** [comment]
Looks like it's mostly working now, just some remaining unit tests that need to be updated for the new error messages: https://github.com/opensearch-project/dashboards-observability/actions/runs/8899395226/job/24438654843?pr=1732#step:6:10907

Again, I recommend running tests locally instead of waiting on actions here, it makes for a faster feedback loop and the tests are typically good at catching these types of small compatibility issues.

**Comment by devesh-2002** [comment]
I tried to change the `JSON5` to `JSON` again in `json_data_adaptor.ts` and when I do `yarn test integrations` the tests are running correctly
```
Test Suites: 23 passed, 23 total
Tests:       154 passed, 154 total
Snapshots:   15 passed, 15 total
Time:        1424.374 s
Ran all test suites matching /integrations/i.
```

**Comment by Swiddis** [comment]
Confirmed that tests are passing after fixing upstream breakage -- will merge this after #1792.

**Review by Swiddis — APPROVED** [review]

**Comment by devesh-2002** [comment]
Thanks a lot for all the suggestions in this PR! @Swiddis @RyanL1997

**Review by ps48 — APPROVED** [review]

**Pull request merged**
